packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts

```diff
@@ -404,6 +403,20 @@ export interface LaunchTemplateProps {
    * @default false
    */
   readonly instanceMetadataTags?: boolean;
+
+  /**
+   * Whether instances should have a public IP addresses associated with them.
+   *
+   * @default - Use subnet settings
+   */
+  readonly associatePublicIpAddress?: boolean;
+
+  /**
+   * The instance profile used to pass role information to EC2 instances.
+   *
```

Comment:
Add a note here about providing either this or role (and do the same for role)

Reply:
Updated.